pkg/scheduler/scheduler.go

```diff
@@ -319,6 +319,35 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 	return entries
 }
 
+// resourcesToReserve calculates how much of the available resources in cq/cohort assignment should be reserved.
+// For Peempt mode it makes sure to not reserve resources more than nominal quota.
+// TODO: we can make reservation more accurate, by accounting how many resources
+// the assignment is going to preempt.
+func resourcesToReserve(e *entry, cq *cache.ClusterQueue) cache.FlavorResourceQuantities {
+	if e.assignment.RepresentativeMode() != flavorassigner.Preempt {
+		return e.assignment.Usage
+	}
+	reservedUsage := make(cache.FlavorResourceQuantities)
+	for flavor, resourceUsage := range e.assignment.Usage {
+		for resource, usage := range resourceUsage {
+			if _, ok := reservedUsage[flavor]; !ok {
+				reservedUsage[flavor] = make(map[corev1.ResourceName]int64)
+			}
```

Comment:
This can be done once in the outer loop, without the conditional, as the flavor is unique in `e.assignment.Usage`.

Reply:
Is the flavor unique? The structure allows to have multiple flavors.

Reply:
It is unique in the `e.assignment.Usage` map.